File: app/src/main/java/top/easelink/lcg/utils/ActivityUtils.java
package top.easelink.lcg.utils;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import top.easelink.framework.base.BaseFragment;

import static com.google.common.base.Preconditions.checkNotNull;

/**
 * This provides methods to help Activities load their UI.
 */
public class ActivityUtils {

    public static final String TAG_PREFIX = "lcg-";
    /**
     * The {@code fragment} is added to the container view with id {@code frameId}. The operation is
     * performed by the {@code fragmentManager}.
     *
     */
    public static void addFragmentInActivity(@NonNull FragmentManager fragmentManager,
                                             @NonNull BaseFragment fragment, int frameId) {
        checkNotNull(fragmentManager);
        checkNotNull(fragment);
        FragmentTransaction transaction =
                fragmentManager
                        .beginTransaction()
//                        .setCustomAnimations(R.anim.slide_left, R.anim.slide_right)
                        .add(frameId, fragment, TAG_PREFIX + fragment.getClass().getSimpleName());
        transaction.commitNow();
    }

    /**
     * The {@code fragment} is added to the container view with id {@code frameId}. The operation is
     * performed by the {@code fragmentManager}.
     *
     */
    public static void replaceFragmentInActivity(@NonNull FragmentManager fragmentManager,
                                                 @NonNull Fragment fragment, String tag) {
        checkNotNull(fragmentManager);
        checkNotNull(fragment);
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.add(fragment, tag);
        transaction.commit();
    }
}

File: app/src/main/java/top/easelink/lcg/utils/ActivityUtils.kt
package top.easelink.lcg.utils

import android.os.Looper
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

/**
 * The `fragment` is added to the container view with id `frameId`. The operation is
 * performed by the `fragmentManager`.
 *
 */
fun addFragmentInActivity(fragmentManager: FragmentManager,
                          fragment: Fragment,
                          frameId: Int) {
    fragmentManager
        .beginTransaction()
        .addToBackStack(fragment.javaClass.simpleName)
        .add(frameId, fragment, fragment.javaClass.simpleName)
        .commitAllowingStateLoss()
}

fun popBackFragmentUntil(fragmentManager: FragmentManager, tag: String): Boolean {
    return fragmentManager.popBackStackImmediate(tag, 0)
}


fun addFragmentInFragment(fragmentManager: FragmentManager, fragment: Fragment, frameId: Int) {
    val transaction = fragmentManager
        .beginTransaction()
        // .setCustomAnimations(R.anim.slide_left, R.anim.slide_right)
        .add(frameId, fragment, fragment.javaClass.simpleName)
    transaction.commitNow()
}

/**
 * The `fragment` is added to the container view with id `frameId`. The operation is
 * performed by the `fragmentManager`.
 *
 */
fun replaceFragmentInActivity(fragmentManager: FragmentManager, fragment: Fragment, frameId: Int) {
    val transaction = fragmentManager
        .beginTransaction()
        .replace(frameId, fragment, fragment.javaClass.simpleName)
    transaction.commitNow()
}

fun isOnMainThread(): Boolean {
    return Looper.myLooper() == Looper.getMainLooper()
}